server/stream.go

```diff
@@ -44,3 +44,22 @@ func (s *server) invokeSendEndpoint() endpoint.SendEndpoint {
 		return stream.SendMsg(req)
 	}
 }
+
+// contextStream is responsible for solving ctx diverge in server side streaming.
+// it receives the ctx from previous middlewares and the Stream that exposed to users，then rewrite
+// Context() method so that users could call Stream.Context() in handler to get the processed ctx.
+type contextStream struct {
```

Comment:
It makes GetServerConn not working AGAIN?

Reply:
Yes, I forgot this part. If users invoke GetServerConn in their handler, it would not work again.